vassal-app/src/main/java/VASSAL/build/module/Chatter.java

```diff
@@ -197,10 +206,74 @@ public void keyReleased(KeyEvent e) {
     add(input);
 
     final Action copyAction = new DefaultEditorKit.CopyAction();
-    copyAction.putValue(Action.NAME, Resources.getString("General.copy"));
+    copyAction.putValue(Action.NAME, Resources.getString("Chat.copy"));
+
+    final Action saveAction = new AbstractAction(Resources.getString("Chat.save_to_html_file")) {
+      @Override
+      public void actionPerformed(ActionEvent e) {
+        final FileChooser fc = GameModule.getGameModule().getFileChooser();
+
+        if (fc.showSaveDialog(GameModule.getGameModule().getControlPanel()) != FileChooser.APPROVE_OPTION) {
+          return;
+        }
+
+        final File sf = fc.getSelectedFile();
+        String name = sf.getPath();
+        final int index = name.lastIndexOf('.');
+        if (index < 0) {
+          name = name + ".html";
+          fc.setSelectedFile(new File(name));
+        }
+
+        final File outputFile = fc.getSelectedFile();
+        if (outputFile == null) {
+          return;
+        }
+
+        try (OutputStream fos = Files.newOutputStream(Path.of(outputFile.getPath()))) {
```

Comment:
`outputFile.toPath()` will get you a `Path`.

Reply:
fixed